pkg/test/e2e/cilium/cilium_test.go

```diff
@@ -0,0 +1,660 @@
+				"pod-to-b-intra-node-nodeport",
+				"pod-to-b-multi-node-clusterip",
+				"pod-to-b-multi-node-headless",
+				"pod-to-b-multi-node-nodeport",
+				"pod-to-external-1111",
+				"pod-to-external-fqdn-allow-google-cnp",
+			}
+
+			pods, err := userClient.CoreV1().Pods(ciliumTestNs).List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				t.Logf("failed to get pod list: %s", err)
+				return false, nil
+			}
+
+			pods.Items = filterByLabel(pods.Items, "name", names...)
+
+			if len(pods.Items) == 0 {
+				t.Logf("no connectivity test pods found")
+				return false, nil
+			}
+
+			allHealthy := true
+			for _, pod := range pods.Items {
+				podHealthy := true
+				if pod.Status.Phase != corev1.PodRunning {
+					podHealthy = false
+					t.Log("not running", pod.Name, pod.Status.Phase)
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == corev1.PodReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not ready", pod.Name, c.Type, c.Status)
+						}
+					} else if c.Type == corev1.ContainersReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not container ready", pod.Name, c.Type, c.Status)
+						}
+					}
+				}
+
+				if !podHealthy {
+					t.Logf("%q not healthy", pod.Name)
+				}
+
+				allHealthy = allHealthy && podHealthy
+			}
+
+			if !allHealthy {
+				t.Logf("not all pods healthy yet...")
+				return false, nil
+			}
+
+			t.Logf("all pods healthy")
+
+			return true, nil
+		})
+		if err != nil {
+			t.Fatalf("pods never became ready: %v", err)
+		}
+	}
+
+	t.Logf("checking for Hubble pods")
+	{
+		err := wait.Poll(30*time.Second, 30*time.Minute, func() (bool, error) {
+			t.Logf("checking pod readiness...")
+
+			names := []string{
+				"hubble-relay",
+				"hubble-ui",
+			}
+
+			pods, err := userClient.CoreV1().Pods("kube-system").List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				t.Logf("failed to get pod list: %s", err)
+				return false, nil
+			}
+
+			pods.Items = filterByLabel(pods.Items, "k8s-app", names...)
+
+			if len(pods.Items) == 0 {
+				t.Logf("no hubble pods found")
+				return false, nil
+			}
+
+			allHealthy := true
+			for _, pod := range pods.Items {
+				podHealthy := true
+				if pod.Status.Phase != corev1.PodRunning {
+					podHealthy = false
+					t.Log("not running", pod.Name, pod.Status.Phase)
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == corev1.PodReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not ready", pod.Name, c.Type, c.Status)
+						}
+					} else if c.Type == corev1.ContainersReady {
+						if c.Status != corev1.ConditionTrue {
+							podHealthy = false
+							t.Log("not container ready", pod.Name, c.Type, c.Status)
+						}
+					}
+				}
+
+				if !podHealthy {
+					t.Logf("%q not healthy", pod.Name)
+				}
+
+				allHealthy = allHealthy && podHealthy
```

Comment:
there are a few places where you are listing pods and checking if they are ready/healthy. Maybe extract that into a function so its not duplicated.